test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1

```diff
@@ -470,6 +470,49 @@ Describe "Additional tests for Import-Module with WinCompat" -Tag "Feature" {
             $LogPath | Should -FileContentMatch 'True'
         }
     }
+
+    Context "Tests around Windows PowerShell Compatibility module deny list" {
+        BeforeAll {
+            $pwsh = "$PSHOME/pwsh"
+            Add-ModulePath $basePath
+            $ConfigPath = Join-Path $TestDrive 'powershell.config.json'
+        }
+
+        AfterAll {
+            Restore-ModulePath
+        }
+
+        It "Successfully imports incompatible module when DenyList is not specified in powershell.config.json" {
+            '{"Microsoft.PowerShell:ExecutionPolicy": "RemoteSigned"}' | Out-File -Force $ConfigPath
+            pwsh -NoProfile -NonInteractive -settingsFile $ConfigPath -c "[System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('TestWindowsPowerShellPSHomeLocation', `'$basePath`');Import-Module $ModuleName2 -WarningAction Ignore;Test-${ModuleName2}PSEdition" | Should -Be 'Desktop'
+        }
+
+        It "Successfully imports incompatible module when DenyList is empty" {
+            '{"WindowsPowerShellCompatibilityModuleDenyList": []}' | Out-File -Force $ConfigPath
+            pwsh -NoProfile -NonInteractive -settingsFile $ConfigPath -c "[System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('TestWindowsPowerShellPSHomeLocation', `'$basePath`');Import-Module $ModuleName2 -WarningAction Ignore;Test-${ModuleName2}PSEdition" | Should -Be 'Desktop'
+        }
+
+        It "Blocks DenyList module import by Import-Module <ModuleName> -UseWindowsPowerShell" {
+            $LogPath = Join-Path $TestDrive (New-Guid).ToString()
+            '{"WindowsPowerShellCompatibilityModuleDenyList": ["' + $ModuleName2 + '"]}' | Out-File -Force $ConfigPath
+            pwsh -NoProfile -NonInteractive -settingsFile $ConfigPath -c "[System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('TestWindowsPowerShellPSHomeLocation', `'$basePath`');Import-Module $ModuleName2 -UseWindowsPowerShell" *> $LogPath
+            $LogPath | Should -FileContentMatch 'blocked from loading'
```

Comment:
Checking for the error message is a problem when the resources get localized.  Should just check against the errorid.

Reply:
Updated.